network.go

```diff
@@ -736,6 +736,15 @@ func (n *network) CreateEndpoint(name string, options ...EndpointOption) (Endpoi
 		ep.ipamOptions[netlabel.MacAddress] = ep.iface.mac.String()
 	}
 
+	if ipam.capability.RequiresEndpointInfo {
+		if ep.ipamOptions == nil {
+			ep.ipamOptions = make(map[string]string)
+		}
+		log.Debugf("Endpoint: %s", ep)
+		ep.ipamOptions[netlabel.EndpointName] = name
```

Comment:
Thanks @johnbelamaric for the explanation.

If this extra information is mainly needed for auditing purposes, then I am suggesting to pass only the endpoint and network ID to the ipam driver.

The IDs are the right thing to pass given they are the only unique identifiers of the network resources, while names are not.

With that information the remote driver would be able to fill in the auditing info for each assigned pool/IP address by querying the daemon.